### DISCUSSION TITLE
--add-data and --add-binary are not able to place a file onto the root folder of the built bundle

### DISCUSSION THREAD

**Discussion by DrBomb** [pyinstaller/pyinstaller#7650] [Help] [answered]
`--add-data` and `--add-binary are` not able to place a file onto the root folder of the built bundle.

Platform: Windows
Running python 3.10, had the same issue with python 3.11

Building with `pyinstaller main.py --add-data text.txt;text.txt` **creates a folder called "text.txt" and inside it, the file "text.txt" is located**. Is that the expected behavior? I've also tried `text.txt;.` and `text.txt;.\` but nothing can place the file on the root of the built bundle.

**Comment by rokm** [comment]
The second (destination) part of the argument is the target directory, not target file name. To place the file into top-level directory, you should use `.` as the target directory, i.e., `--add-data text.txt;.`.

**Comment by DrBomb** [comment]
Could've sworn I tried this!

Seems like setting it up incorrectly the first time with `--add-data text.txt;text.txt` and then trying to fix it with `--add-data text.txt;.` will not work unless the `build` folder is deleted beforehand. Weird given that the .spec file will update with the latest data paths you're giving it.

I don't know if it is worth opening an issue for it. But deleting `build` did the trick, thanks for the answer, I appreciate it.

**Reply by rokm** [reply]
> Could've sworn I tried this!
> 
> Seems like setting it up incorrectly the first time with `--add-data text.txt;text.txt` and then trying to fix it with `--add-data text.txt;.` will not work unless the `build` folder is deleted beforehand. Weird given that the .spec file will update with the latest data paths you're giving it.
> 
> I don't know if it is worth opening an issue for it. But deleting `build` did the trick, thanks for the answer, I appreciate it.

Thanks for letting us know! That's the second indication today that data-related caching might be misbehaving. No need to open an issue, though, I'm planning to look at it anyway.

**Reply by rokm** [reply, marked as answer]
Should be fixed by #7653.

**Comment by DrBomb** [comment]
Awesome, thank you
